src/styles/_catalog.scss

```diff
@@ -1,22 +1,40 @@
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  grid-auto-flow: row;
+  justify-content: center;
+  grid-template-columns: repeat(4, $card-size);
+  gap: $gap;
 
-  max-width: 944px;
-  margin: 0 auto;
+  @media screen and (min-width:300px) and ( max-width:487.99px) {
+    grid-template-columns: repeat(1, $card-size);
+  }
+
+  @media screen and (min-width:488px) and ( max-width:767.99px) {
+    grid-template-columns: repeat(2, $card-size);
+  }
+
+  @media screen and (min-width:768px) and ( max-width:1023.99px) {
+    grid-template-columns: repeat(3, $card-size);
+  }
+
+  @media screen and ( min-width:1024px) and (max-width:4096px) {
+    grid-template-columns: repeat(4, $card-size);
+  }
```

Comment:
good oportunity to use a mixin here.  also when you do a media query, there is no need to use both `min-width` and `max-width`, choose either one